File: frontend/steward-ui/src/lib/models/policies/OverridePolicy.ts
import type { Allowance } from "..";

// TODO: Remove override policy completely. Allowed will completely depend on whether an OverrideAllowance is provided.
// Move optional requirement and allowance (which should also become optional and renamed to OverrideAccess) to the policy directly.
// A null `OverrideRequirement` means that nothing is required to submit an override request.

export interface OverridePolicy {
    allowed: boolean;
    requirement?: OverrideRequirement;
    allowance: Allowance;
}

export type OverrideRequirement =
    | "delay"
    | "randomText"
    | "userApproval";

File: src/Steward.Server/Data/Policies/OverridePolicy.cs
namespace Steward.Server.Data.Policies;

public class OverridePolicy
{
    public bool Allowed { get; set; }

    public OverrideRequirement? Requirement { get; set; }

    public Allowance Allowance { get; set; } = new();
}

public enum OverrideRequirement
{
    Delay,
    RandomText,
    UserApproval
}